src/content/interop/js-interop/package-web.md

```diff
@@ -0,0 +1,307 @@
+as the class that it annotates.
+
+Similarly, if you find an API with the keyword `native` in `dart:html` that
+doesn't have an equivalent in `package:web`, check to see if there was a rename
+with the `@JSName` annotation.
+The annotation tells the compiler to use the value in the annotation instead of
+the member name, much like [`@JS`] does.
+`native` is an internal keyword that means the same as `external` in this
+context.
+
+### Type checks
+
+It's common for code that uses `dart:html` to utilize runtime checks like `is`.
+When used with a `dart:html` object, `is` and `as` verify that the object is
+the JS type within the `@Native` annotation.
+In contrast, all `package:web` types are reified to [`JSObject`][]. This means a
+runtime type check will result in different behavior between `dart:html` and
+`package:web` types.
+
+To be able to perform type checks, migrate any `dart:html` code
+using `is` type tests to use [interop methods][] like `instanceOfString`.
+The [Compatibility, type checks, and casts][]
+section of the JS types page covers alternatives in detail.
+
+```dart
+obj is Window; // Remove
+obj.instanceOfString('Window'); // Add
+```
+
+### Type signatures
+
+Many APIs in `dart:html` support various Dart types in their type signatures.
+Because `dart:js_interop` [restricts] the types that can be written, some of
+the members in `package:web` will now require you to *convert* the value before
+calling the member.
+Learn how to use interop conversion methods from the [Conversions][]
+section of the JS types page.
+
+```dart
+window.addEventListener('click', callback); // Remove
+window.addEventListener('click', callback.toJS); // Add
+```
+
+{% comment %}
+TODO: Think of a better example. People will likely use the stream helpers
+instead of `addEventListener`.
+{% endcomment -%}
+
+Generally, you can spot which methods need a conversion because they'll be
+flagged with some variation of the exception:
+
+```plaintext
+A value of type '...' can't be assigned to a variable of type 'JSFunction?'
+```
+
+### Conditional imports
+
+It is common for code to use a conditional import based on whether `dart:html`
+is supported to differentiate between native and web:
+
+```dart
+export 'src/hw_none.dart'
+    if (dart.library.io) 'src/hw_io.dart'
+    if (dart.library.html) 'src/hw_html.dart';
+```
+
+However, since `dart:html` is not supported when compiling to Wasm, the correct
+alternative now is to use `dart.library.js_interop` to differentiate between
+native and web:
+
+```dart
+export 'src/hw_none.dart'
+    if (dart.library.io) 'src/hw_io.dart'
+    if (dart.library.js_interop) 'src/hw_web.dart';
+```
+
+### Virtual dispatch and mocking
+
+`dart:html` classes supported virtual dispatch, whereas because JS interop uses
+extension types, virtual dispatch is [not possible]. Similarly, `dynamic` calls
+with `package:web` types won't work as expected as their members are only
```

Comment:
well. there is a caveat: if `dart:html` is included in the SDK, dart2js will continue to find the interceptor and it may continue to work by accident. Once `dart:html` is removed it will stop working.

Not sure if it's worth flagging this, since it's a very confusing concept to explain.

Reply:
I added a caveat without getting into detail:

> ...won't work as expected (or, they might continue to work
just by chance, but will stop when `dart:html` is removed), ...

Reply:
> Not sure if it's worth flagging this, since it's a very confusing concept to explain.

I had a comment for this and then removed it figuring it would just confuse users, but either way is fine. :)